akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/PropsAdapter.scala

```diff
@@ -21,11 +23,22 @@ import akka.annotation.InternalApi
       rethrowTypedFailure: Boolean = true): akka.actor.Props = {
     val props = akka.actor.Props(new ActorAdapter(behavior(), rethrowTypedFailure))
 
-    (deploy.firstOrElse[DispatcherSelector](DispatcherDefault.empty) match {
+    val p1 = (deploy.firstOrElse[DispatcherSelector](DispatcherDefault.empty) match {
       case _: DispatcherDefault          => props
       case DispatcherFromConfig(name, _) => props.withDispatcher(name)
       case _: DispatcherSameAsParent     => props.withDispatcher(Deploy.DispatcherSameAsParent)
     }).withDeploy(Deploy.local) // disallow remote deployment for typed actors
+
+    val p2 = deploy.firstOrElse[MailboxSelector](MailboxSelector.defaultMailbox()) match {
```

Comment:
Maybe use Scala API here?

```suggestion
    val p2 = deploy.firstOrElse[MailboxSelector](MailboxSelector.default()) match {
```